tests/unit/Cassandra/DurationTest.php

```diff
@@ -49,6 +49,78 @@ public function testStringArgParseError()
     }
 
     /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage nanos must be between -2147483648 and 2147483647, 8589934592 given
+     */
+    public function testStringArgOverflowError()
+    {
+        new Duration(1, 2, "8589934592");
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage nanos must be between -2147483648 and 2147483647, -8589934592 given
+     */
+    public function testStringArgUnderflowError()
+    {
+        new Duration(1, 2, "-8589934592");
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage nanos must be between -2147483648 and 2147483647, 8589934592 given
+     */
+    public function testBigintArgOverflowError()
+    {
+        new Duration(1, 2, new Bigint("8589934592"));
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage nanos must be between -2147483648 and 2147483647, -8589934592 given
+     */
+    public function testBigintArgUnderflowError()
+    {
+        new Duration(1, 2, new Bigint("-8589934592"));
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessageRegExp /days must be between -2147483648 and 2147483647, 8\.?58993.* given/
```

Comment:
In this case, the exception message varies among platforms: the "given" is either formatted in scientific notation (on most Windows / VC flavors) or straight out (on Linux).